pkg/cloud/vsphere/vsphere_cloud_config/config_test.go

```diff
@@ -0,0 +1,182 @@
+package vsphere_cloud_config
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	gmg "github.com/onsi/gomega"
+)
+
+const basicConfigINI = `
+[Global]
+server = 0.0.0.0
+port = 443
+user = user
+password = password
+insecure-flag = true
+datacenters = us-west
+ca-file = /some/path/to/a/ca.pem
+`
+
+const basicConfigYaml = `
+global:
+  user: user
+  password: password
+  server: 0.0.0.0
+  port: 443
+  insecureFlag: true
+  datacenters:
+  - us-west
+  caFile: /some/path/to/a/ca.pem
+`
+
+const basicConfigVcenterSectionINI = `
+[Global]
+secret-name = "global-secret"
+secret-namespace = "global-secret-ns"
+
+[VirtualCenter "vc.rh.com"]
+datacenters = "DC0,DC1"
+
+[Labels]
+region = "k8s-region"
+zone = "k8s-zone"
+`
+
+const basicConfigVcenterSectionYAML = `
+global:
+  secretName: global-secret
+  secretNamespace: global-secret-ns
+vcenter:
+  vc.rh.com:
+    server: vc.rh.com
+    datacenters:
+    - DC0
+    - DC1
+labels:
+  zone: k8s-zone
+  region: k8s-region
+`
+
+const multiVCDCsConfigINI = `
+[Global]
+port = 443
+insecure-flag = true
+secret-name = "global-secret"
+secret-namespace = "global-secret-ns"
+
+[VirtualCenter "t1"]
+server = "10.0.0.1"
+datacenters = "DC0,DC1,DC2"
+secret-name = "tenant1-secret"
+secret-namespace = "kube-system"
+
+[VirtualCenter "10.0.0.2"]
+datacenters = "DC3"
+
+[VirtualCenter "10.0.0.3"]
+datacenters = "DC5,DC6"
+ip-family = "ipv6"
+`
+
+const invalidConfig = `boom[]{}`
+
+const invalidConfigWrongGlobalPort = `
+[Global]
+port = -443
+insecure-flag = true
+`
+
+const invalidConfigWrongVCPort = `
+[Global]
+port = 443
+insecure-flag = true
+
+[VirtualCenter "10.0.0.3"]
+datacenters = "DC5,DC6"
+port = -1
+ip-family = "ipv6"
+`
+
+func TestINIConfigConversion(t *testing.T) {
+
+	t.Run("basic config yaml conversion", func(t *testing.T) {
+		g := gmg.NewWithT(t)
+		configStruct, err := ReadConfig([]byte(basicConfigINI))
+		g.Expect(err).Should(gmg.Succeed())
+
+		convertedConfig, err := MarshalConfig(configStruct)
+		g.Expect(err).Should(gmg.Succeed())
```

Comment:
As above, error checks should be "to not have occurred", please fix the other occurrences too
```suggestion
		g.Expect(err).ToNot(gmg.HaveOccurred())
```